src/main/java/com/github/princesslana/eriscasper/gateway/Gateway.java

```diff
@@ -161,10 +161,10 @@ private Completable resume(RxWebSocket ws, BotToken token) {
         lastSeenSequenceNumber.isPresent(), "Can not resume without a sequence number");
 
     return Single.just(
-            ImmutableResume.builder()
-                .token(token)
-                .sessionId(sessionId.get())
-                .seq(lastSeenSequenceNumber.get())
+            ImmutableResumePayload.builder()
+                .token(token.unwrap())
+                .sessionId(sessionId.get().unwrap())
```

Comment:
I realize get was used before, but what are we calling `get` on here? A nullable or optional? Can we do better?

Reply:
Optional, we can do better, I replaced it with a much better system in my opinion with better errors thrown.